### DISCUSSION TITLE
Install specific version of Hugo via Homebrew

### DISCUSSION THREAD

**Discussion by marksou** [Homebrew/discussions#2941] [Getting started] [answered]
I've been trying to help a co-worker install the specific version of Hugo we use. When running `brew install hugo@0.64.1` it comes back saying it can't find the formula. Is there anything else that needs to be done or do we need to find an alternative method?

**Comment by SMillerDev** [comment]
Homebrew-core only ships the latest version of software so this won't work. It might be beneficial for your company to use `brew extract` to extract the specific version you need into a [tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap) so all colleagues can use the same versions.

**Comment by carlocab** [comment, marked as answer]
`brew` doesn't let you install arbitrary versions of formulae out of the box, I'm afraid. What you can do is the following:
```
brew tap-new marksou/your-tap-name-here # create your own tap if you don't already have one
brew extract --version=0.64.1 hugo marksou/your-tap-name-here # extract the version of Hugo that you need to your tap
brew install marksou/your-tap-name-here/hugo@0.64.1
```
You can also modify your formula before installing it with
```
brew edit marksou/your-tap-name-here/hugo@0.64.1
```

**Reply by marksou** [reply]
Thanks

**Reply by carlocab** [reply]
Sure. If you have multiple users using the same version, you may wish to push your tap to GitHub (a tap is just a Git repo).

You can find the repo at `$(brew --repo marksou/your-tap-name-here)`. You'll want to create a repository called `homebrew-your-tap-name-here` and push your tap there.

Then, other users can install the version in your tap with
```
brew install marksou/your-tap-name-here/hugo@0.64.1
```

Here's some more relevant documentation:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Reply by TheAbstract** [reply]
@carlocab Thanks, very helpful. Saved from having to find `*.rb` files in Formula from commit histories 😌, as mentioned in most answers to this online. 
FYI: for those that don't need create a tap, the required steps can be simplified like this:
```
brew extract --version=0.51 hugo homebrew/cask
brew install hugo@0.51
```
**Note:** unless it's just for personal use, I highly recommend creating a tap as described above https://github.com/Homebrew/discussions/discussions/2941?sort=top#discussioncomment-2155711.

**Comment by chriszrc** [comment]
@carlocab these instructions didn't work for me on a brand new MacBook with Monterey. It appears homebrew was trying to build hugo from source, and running into all kinds of errors. What worked was to install the binary from github:

https://github.com/gohugoio/hugo/releases/tag/v0.79.0

Following these instructions - https://datacenterjourney.com/2021/install-specific-version-hugo-macos/

**Comment by peterwake** [comment]
I just had this problem and wrote a gist on how I fixed it here - https://gist.github.com/peterwake/2851767e4424d11688d9a6f40149c3da

Hope it's useful to someone!
